lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
AFAIK, the purpose of calling `valid?` here is to run the validations, not so much to actually use its return value. Without `valid?`, `errors` wouldn't be populated.